src/RequestsLibrary/RequestsOnSessionKeywords.py

```diff
@@ -167,14 +167,14 @@ def head_on_session(self, alias, url,
         see the `GET` keyword for the complete list.
         """
         session = self._cache.switch(alias)
-        response = self._common_request("head", session, url, **kwargs)
+        response = self._common_request("HEAD", session, url, allow_redirects=allow_redirects, **kwargs)
         self._check_status(expected_status, response, msg)
         return response
 
     @keyword("OPTIONS On Session")
     @warn_if_equal_symbol_in_url_on_session
     def options_on_session(self, alias, url,
-                           expected_status=None, msg=None, **kwargs):
+                           expected_status=None, msg=None, allow_redirects=False, **kwargs):
```

Comment:
Same here

Reply:
As mentioned in the documentation, and the existing acceptance tests. Redirection should be disabled by default for these methods.

Reply:
I dug a little deeper:
This does not redirect by default:
`>>> x=requests.head("http://localhost:5010/redirect-to?url=anything")`
`>>> x`
`<Response [302]>`

This does redirect by default:
`>>> x=requests.request("HEAD","http://localhost:5010/redirect-to?url=anything")`
`>>> x`
`<Response [200]>`

So my change changed the default behavior of the HEAD and HEAD on session keywords. Ill make sure that only the HEAD keywords don't redirect by default.

Reply:
I think would be better to leave the default values to the upstream requests library itself instead of handling them in this library.

Reply:
Ok thats fine, but that will change the default behavior for HEAD requests

Reply:
Ah ok now I understand, wondering why and what's better.
Maybe we can open an upstream issue and see what they say and what method is better to use to make requests.

Reply:
Before the requests.get/post/delete etc methods were used. But there are no methods for trace and connect http methods upstream. That's why I switched to the generic request method. But to keep the same behavior for HEAD I had to add a default allow_redirection value. Ill make an issue upstream.

Reply:
> Before the requests.get/post/delete etc methods were used. But there are no methods for trace and connect http methods upstream. That's why I switched to the generic request method. But to keep the same behavior for HEAD I had to add a default allow_redirection value. Ill make an issue upstream.

Yep I understood.

Reply:
https://github.com/psf/requests/issues/6723

Reply:
The upstream default allow_redirects value for head requests is as intended. Also there is no intention to add support for TRACE and CONNECT methods upstream. We can only handle it in this library.